dicons.c

```diff
@@ -154,22 +171,35 @@ static GtkListStore *create_desktop_list(void)
     return GTK_LIST_STORE(store);
 }
 
+static void launch_desktop_shortcut(GFile *desktop_file) {
```

Comment:
function name is misleading. the files on the desktop are not necessarily shortcuts. maybe something like `launch_default_or_app_for_file` (isnt very beautiful either but ehhh; maybe you come up with something better).